Source/Particles/Collision/BinaryCollision/BinaryCollision.H

```diff
@@ -199,6 +217,10 @@ public:
                     amrex::HostDevice::Atomic::Add( &(*cost)[mfi.index()], wt);
                 }
             }
+
+            // Call redistribute to remove particles with negative ids
+            species1.Redistribute(lev, lev, 0, true, true);
```

Comment:
Is this always needed? We should know whether collisions might remove particles.

Reply:
As discussed with @dpgrote, we could add a flag (known at compilation type) checking whether this type of collision can remove particles. `Redistribute` would only need to be called in that case.

Reply:
Good point! Any collision that has product species would be capable of removing particles, so we can simply wrap this in a 
```
if (m_have_product_species) {
    ...
}
```

Btw, this change came in with #4744 so it might be better to review (and merge) that PR before reviewing this one too much further.

Reply:
I updated this in the other PR.